### PR TITLE
feat!: add ability to copy and paste workspace comments

### LINKED ISSUE #8008: Wire together copy paste for workspace comments

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes #8008 

### Proposed Changes

Integrates the new `RenderedWorkspaceComment` with the copy paste system.

### Test Coverage

Manually tested that copying and pasting works properly. Including with text, collapsed comments, resized comments, etc.

### Documentation

N/A

### Additional Information

N/A

### CODE REVIEW

**Comment by BeksOmega** [comment]
@gonfunko This isn't blocking the beta release on Monday so don't stress if you're busy today. But I am planning to build on top of this code.

File: core/clipboard/workspace_comment_paster.ts [hunk 1]
```diff
@@ -8,38 +8,87 @@ import {IPaster} from '../interfaces/i_paster.js';
...
+
+function moveCommentToNotConflict(comment: RenderedWorkspaceComment) {
+  const workspace = comment.workspace;
+  const translateDistance = 30;
+  const coord = comment.getRelativeToSurfaceXY();
+  const offset = new Coordinate(0, 0);
+  // getRelativeToSurfaceXY is really expensive, so we want to cache this.
+  const otherCoords = workspace
+    .getTopComments(false)
+    .filter((otherComment) => otherComment.id != comment.id)
```
Comment (gonfunko) [inline core/clipboard/workspace_comment_paster.ts:64]: Probably want !== here?

File: core/clipboard/workspace_comment_paster.ts [hunk 2]
```diff
@@ -8,38 +8,87 @@ import {IPaster} from '../interfaces/i_paster.js';
...
+function commentOverlapsOtherExactly(
+  coord: Coordinate,
+  otherCoords: Coordinate[],
+): boolean {
+  return otherCoords.some(
+    (otherCoord) =>
+      Math.abs(otherCoord.x - coord.x) <= 1 &&
+      Math.abs(otherCoord.y - coord.y) <= 1,
+  );
 }
```
Comment (gonfunko) [inline core/clipboard/workspace_comment_paster.ts:88]: WDYT about adding this to the Comment class? Seems like it's generically applicable and might be better there than here.

Comment (BeksOmega) [reply core/clipboard/workspace_comment_paster.ts:88]: Hmm we have a similar function for blocks in the block paster that I don't think we've used anywhere else. So for now I'd rather just keep it internal to this module.

Comment (gonfunko) [reply core/clipboard/workspace_comment_paster.ts:88]: SGTM

**Review by gonfunko — APPROVED** [review]

**Pull request merged**